apps/api/src/app/events/usecases/process-subscriber/process-subscriber.usecase.ts

```diff
@@ -113,14 +102,30 @@ export class ProcessSubscriber {
         _templateId: notification._templateId,
         digest: step.metadata,
         type: step.template.type,
-        providerId: integration?.providerId ?? InAppProviderIdEnum.Novu,
+        providerId: providerId,
         ...(actorSubscriber && { _actorId: actorSubscriber._id }),
       });
     }
 
     return jobs;
   }
 
+  private async getProviderId(command: ProcessSubscriberCommand, stepType: StepTypeEnum): Promise<string | null> {
+    const channelType = ChannelTypeEnum[stepType.toUpperCase()];
+    if (!channelType) return null;
```

Comment:
I feel like this might bring bugs here, but I'm not sure before we were always settings the `providerId` as `InAppProviderIdEnum.Novu` and now it can be null. And the actual type for `providerId` is `string | undefined`.

Reply:
For Null/undefined cases storing InAppProviderIdEnum.Novu does not make sense right? It is saving null fine for delay step, why you think it creates problem? should I change to undefined instead of null?

Reply:
It creates the TS error in line: 105, please change it to `undefined` and the return type to `Promise<string | undefined>`.
I've checked the rest of the code and it seems like it should be fine with `undefined`.